site/blog/2016/jul/1/sphinx-and-rtd-for-writers.rst

```diff
@@ -0,0 +1,548 @@
+In the ``figure``,
+``screenshot-control-panel.jpg`` is an *argument*,
+``:width:`` is an *option*,
+and the rest is the *content*
+They allow for customization of directives,
+and show the full power of reStructuredText.
+
+.. note:: You'll notice that Sphinx uses whitespace to denote where a directive ends.
+   The "Proceed with caution!" is still part of the ``warning``,
+   and everything that continues to be indented will be part of that warning.
+
+Sphinx ships with a number of powerful directives,
+and you can also write your own if you have someone on your team that knows Python.
+
+Inline Markup
+`````````````
+
+For extendability inside of your prose Sphinx uses roles.
+Here is an example:
+
+    You can learn more about this in :rfc:`1984`.
+    It is implemented in our code at :class:`System.Security`
+
+Here you can see ``rfc`` and ``class`` act as the name of the role,
+and then you can pass in a single argument.
+For the ``rfc`` role,
+it generates a link to the online reference for RFC 1984 with a text of ``RFC 1984``.
+
+The ``class`` is where things get interesting.
+This acts as a reference to the class ``System.Security`` that is documented in your project,
+which is a hyperlink in the HTML output,
+but also a working link in PDF and other outputs as well.
+
+Similarly,
+you can implement your own roles with Python.
+This allows you to create powerful semantic constructs into the actual *markup* you're using to write documentation.
+
+Now,
+let's see how you take these sets of reStructuredText files and turn them into a document set.
+
+Table of Contents Tree
+----------------------
+
+One of the main concepts in Sphinx is that it allows multiple pages to be combined into a cohesive hierarchy.
+The ``toctree`` directive is a fundamental part of this structure.
+
+.. code-block:: rst
+
+  .. toctree::
+     :maxdepth: 2
+
+     install
+     support
+
+The above example outputs a Table of Contents in the page where it occurs.
+The ``maxdepth`` argument tells Sphinx to include 2 levels of headers.
+This also tells Sphinx that the other pages are sub-pages of the current page,
+creating a "tree" structure of the pages::
+
+    index
+    ├── install
+    ├── support
+
+The TOC Tree is also used for generating the global navigation inside of Sphinx.
+The `index` at the top of your project acts as the root of the navigation.
+The ``toctree`` is quite important,
+and one of the most powerful concepts in Sphinx.
+
+More info on the ``toctree`` can be found here:
+
+* http://www.sphinx-doc.org/en/stable/markup/toctree.html
+
+Working with Code
+`````````````````
+
+Showing code examples is a common task in all documentation.
+Sphinx is quite well prepared for this task.
+Let's see how we can show a basic code example::
+
+    .. code-block:: python
+       :linenos:
+
+       import antigravity
+
+       def main():
+           antigravity.fly()
+
+This would display the Python snippet,
+syntax highlighted,
```

Comment:
"This markup displays the Python snippet with syntax highlighting and line numbers." ("would" awa "will" ;) -- but there aren't as many of 'em)